packages/util/src/internal.ts

```diff
@@ -0,0 +1,224 @@
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE
+ */
+
+/**
+ * Returns a `Boolean` on whether or not the a `String` starts with '0x'
+ * @param str the string input value
+ * @return a boolean if it is or is not hex prefixed
+ * @throws if the str input is not a string
+ */
+export function isHexPrefixed(str: string): boolean {
+  if (typeof str !== 'string') {
+    throw new Error(
+      "[is-hex-prefixed] value must be type 'string', is currently type " +
+        typeof str +
+        ', while checking isHexPrefixed.'
+    )
+  }
+
+  return str.slice(0, 2) === '0x'
+}
+
+/**
+ * Removes '0x' from a given `String` if present
+ * @param str the string value
+ * @return  a string by pass if necessary
+ */
+export const stripHexPrefix = (str: string): string => {
+  if (typeof str !== 'string') return str
+
+  return isHexPrefixed(str) ? str.slice(2) : str
+}
+
+/**
+ * Pads a `String` to have an even length
+ * @param value
+ * @return output
+ */
+export function padToEven(value: string) {
+  let a = value
+
+  if (typeof a !== 'string') {
+    throw new Error(
+      `[ethjs-util] while padding to even, value must be string, is currently ${typeof a}, while padToEven.`
+    )
+  }
+
+  if (a.length % 2) a = `0${a}`
+
+  return a
+}
+
+/**
+ * Get the binary size of a string
+ * @param str
+ * @return
+ */
+export function getBinarySize(str: string) {
+  if (typeof str !== 'string') {
+    throw new Error(
+      `[ethjs-util] while getting binary size, method getBinarySize requires input 'str' to be type String, got '${typeof str}'.`
+    )
+  }
+
+  return Buffer.byteLength(str, 'utf8')
+}
+
+/**
+ * Returns TRUE if the first specified array contains all elements
+ * from the second one. FALSE otherwise.
+ *
+ * @param superset
+ * @param subset
+ *
+ */
+export function arrayContainsArray(
+  superset: unknown[],
+  subset: unknown[],
+  some?: boolean
+): boolean {
+  if (Array.isArray(superset) !== true) {
+    throw new Error(
+      `[ethjs-util] method arrayContainsArray requires input 'superset' to be an array got type '${typeof superset}'`
+    )
+  }
+  if (Array.isArray(subset) !== true) {
+    throw new Error(
+      `[ethjs-util] method arrayContainsArray requires input 'subset' to be an array got type '${typeof subset}'`
+    )
+  }
+
+  return subset[(Boolean(some) && 'some') || 'every']((value) => superset.indexOf(value) >= 0)
```

Comment:
hard for me to read/parse this if it can be improved/simplified at all

Reply:
@ryanio this was in ethjs-util source

Reply:
that's fine, was just making a comment about it